### ISSUE TITLE
Can Scratchx Run on a Local Webserver?

### ISSUE THREAD

**Issue by CreativeComputerLab** [LLK/scratchx#38]
I require an offline version of Scratchx due to my work in rural locations. Is it possible for
me to host a version of Scratchx on my own local web server? Is the file structure of this
repository such that I can clone them to a location where I can point a web server to serve them?
If so, I'd like to host them using a simple NodeJs server. Are there any special things to consider as this
is a Flash app?

**Comment by cwillisf** [comment]
For the most part, ScratchX is self-contained. There are two main exceptions:
- Costumes and backdrops from the media library are loaded from the main Scratch site. You can use your own costumes and backdrops with the `Upload Sprite from File` button (and others like it) in order to work around that.
- In order for ScratchX to load files from some other site, that site must have a `crossdomain.xml` file that lists the site hosting the SWF. See where `scratchx.org` is mentioned [here](https://github.com/LLK/scratchx/wiki#user-content-setting-up-crossdomainxml).

In short, a copy of ScratchX hosted elsewhere would probably be fine for local development work, but would not have access to as much content (costumes, backdrops, and SBX files) as the main ScratchX site.

**Comment by CreativeComputerLab** [comment]
Thanks. I'll ask about loading extensions from the local file system in a separate thread.
